homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <stdexcept>
+
+template <typename T>
+class unique_ptr {
+public:
+    unique_ptr(T* ptr);
+    unique_ptr(const unique_ptr&) = delete;
+    unique_ptr(unique_ptr&& otherPtr);
+    ~unique_ptr();
+
+    T& operator*() const;
+    T* operator->() const;
+    T* get() const;
+    T* release();
+    void reset(T* newPtr);
```

Comment:
```suggestion
    void reset(T* newPtr = nullptr);
```
default argument would be useful

Reply:
done